pytorch_lightning/core/lightning.py

```diff
@@ -1493,6 +1493,42 @@ def configure_gradient_clipping(self, optimizer, optimizer_idx, gradient_clip_va
             optimizer, gradient_clip_val=gradient_clip_val, gradient_clip_algorithm=gradient_clip_algorithm
         )
 
+    def lr_scheduler_step(
+        self,
+        scheduler: Any,
+        optimizer_idx: Optional[int] = None,
```

Comment:
Should this be always int?

Reply:
ideally yes... there should be.. but doesn't seem to be the case on master even with multiple optimizers. There might be an underlying bug here. Will investigate.

Reply:
yes: https://github.com/PyTorchLightning/pytorch-lightning/pull/11247

Reply:
Passing always int for now with `or 0`, we do the type narrowing in another PR